apps/nsfw-detection/app/nsfw-detection.py

```diff
@@ -0,0 +1,195 @@
+
+    """
+        Generates n FindImage Queries
+    """
+
+    def __init__(self, db, max_images=0, batch_size=16):
+
+        self.db = db
+
+        self.detector = NSFWDetector("input/saved_model.h5")
+
+        query = [{
+            "FindImage": {
+                "constraints": {
+                    "wf_nsfw_detector_neutral": ["==", None]
+                },
+                "results": {
+                    "count": True
+                }
+            }
+        }]
+
+        response, _ = db.query(query)
+
+        try:
+            total_images = response[0]["FindImage"]["count"]
+        except:
+            print("Error retrieving the number of images. No images in the db?")
+            exit(0)
+
+        if total_images == 0:
+            print("No images in the database. Bye!")
+            exit(0)
+
+        print(f"total_images: {total_images}")
+
+        self.batch_size = batch_size
+        self.total_batches = int(total_images / self.batch_size)
+
+        if max_images > 0:
+            self.len = min(self.total_batches, max_images//self.batch_size)
+
+    def __len__(self):
+        return self.len
+
+    def getitem(self, idx):
+
+        if idx < 0 or self.len <= idx:
+            return None
+
+        query = [{
+            "FindImage": {
+                "blobs": True,
+                "constraints": {
+                    "wf_nsfw_detector_neutral": ["==", None]
+                },
+                "batch": {
+                    "batch_size": self.batch_size,
+                    "batch_id": idx
+                },
+                "operations": [
+                    {
+                        "type": "resize",
+                        "width": 224,
+                        "height": 224
+                    }
+                ],
+                "results": {
+                    "list": ["_uniqueid"]
+                }
+            }
+        }]
+
+        return query, []
+
+    def response_handler(self, query, blobs, response, r_blobs):
+
+        try:
+            uniqueids = [i["_uniqueid"] for i in response[0]["FindImage"]["entities"]]
+        except:
+            print(f"error: {response}")
+            return 0
+
+        predictions = self.detector.infer_batch(r_blobs)
+
+        query = []
+
+        for prediction, uniqueid in zip(predictions, uniqueids):
+
+            query.append({
+                "UpdateImage": {
+                    "constraints": {
+                        "_uniqueid": ["==", uniqueid]
+                    },
+                    "properties": {
+                        "wf_nsfw_detector_neutral": prediction["neutral"],
+                        "wf_nsfw_detector_hentai": prediction["hentai"],
+                        "wf_nsfw_detector_porn": prediction["porn"],
+                        "wf_nsfw_detector_sexy": prediction["sexy"],
+                    }
+                }
+            })
+
+        db = self.db.clone()
+
+        r, _ = db.query(query)
```

Comment:
This can be replaced with CommonLibrary.execute_query(db, query). Doing that would be convenient and you can remove the following check for las_query_ok.